### DISCUSSION TITLE
[Feature] Suggestion: Some folders can exclude facial recognition

### DISCUSSION THREAD

**Discussion by nodis** [immich-app/immich#9089] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Suggestion: Some folders can exclude facial recognition, such as the scenic image Kush, which also has no human faces. This can not only accelerate recognition speed but also prevent misidentification

### Platform

- [ ] Server
- [ ] Web
- [ ] Mobile

**Comment by grunwalski** [comment]
Yes. Especially with external libraries it would be really nice to have a blacklist or the option to filter stuff, like in the general scan settings of a library.

**Comment by PetitMote** [comment]
Hello there,
I agree with you, I was coming on Github to ask the same thing :) I think it’d be great if we could deactivate facial recognition per user or per folder.

To explain my personal problem:
I use digiKam to edit my photo library, as I prefer to have everything inside the metadata. I’m synchronizing using Nextcloud to an External Library. When Immich scans the library, it runs face detection agains images with no face included in metadata (but I’ve already checked all my photos). Also, Immich is clustering again the faces, despite having a name for each one. That means I’m having a named person splitted into 2 people with the same name.

I’ve disabled Facial Recognition to avoid this, which works great, but I’d like to open my Immich to my friends and let them use Facial Recognition.

By the way, thank you everybody for Immich, I think it’s really great and comfortable to use ^^

**Comment by Epithumia** [comment]
Similar use case here: I have folders in my external libraries that are of demonstrations, so I end up with thousands of faces of anonymous (to me) people that don't need facial recognition in the first place. If I could avoid having to manually hide 4k+ people, that would be great.

**Comment by Jindra-Dev04** [comment]
Hi, 
My use case: I want to have my meme folder backed up, and it detects like 400+ faces.
It would be great if I could disable face recognition for that folder.

**Comment by mmomjian** [comment]
This FR can be used to track disabling any/all ML features from a library and/or user (smart search, facial recognition, etc)

#9961

**Comment by DawidKrynski** [comment]
I had the same problem with a ~1,900-photo party album that filled People/Explore with hundreds of strangers. Excluding the album from face recognition wasn't quite what I wanted, though. If someone from it also appears in my own photos, I still want them recognised.

So I built a small standalone tool that hides a person only if **every** photo of them is in the excluded albums: https://github.com/DawidKrynski/immich-album-people-hider

It uses only the public API (v3.2+), is dry-run by default, only toggles `isHidden`, remembers whom it hid (and unhides them if they later appear elsewhere), and never touches people you hid or unhid manually. On my library it hid 175 party-only people out of ~750, and everyone who also appears in my own photos stayed visible.

Maybe useful to others until there's a native option.
